packages/profile-sync-controller/src/shared/encryption/utils.ts

```diff
@@ -14,3 +16,35 @@ export const bytesToUtf8 = (byteArray: Uint8Array) => {
 export const stringToByteArray = (str: string) => {
   return new TextEncoder().encode(str);
 };
+
+export const areAllUInt8ArraysUnique = (arrays: Uint8Array[]) => {
+  const seen = new Set<string>();
+  for (const arr of arrays) {
+    const str = arr.toString();
+    if (seen.has(str)) {
+      return false;
+    }
+    seen.add(str);
+  }
+  return true;
+};
+
+/**
+ * Returns a boolean indicating if the entries have different salts.
+ *
+ * @param entries - User Storage Entries
+ * @returns A boolean indicating if the entries have different salts.
+ */
+export function getIfEntriesHaveDifferentSalts(entries: string[]): boolean {
+  const salts = entries
+    .map((e) => {
+      try {
+        return encryption.getSalt(e);
```

Comment:
We also need them for SDK tests, in order to avoid duplication and / or import spillage, I've put them in the shared folder. But you're right, this creates a circular dep. Should we assume this would get tree-shaken?

Anyway, I've moved this util function directly inside `encryption.ts` and added the tests in `encryption.test.ts`!

Reply:
Haha, I would assume tree shaking... But we are using this in extension and mobile lol.

You could validate by making a preview build, and testing on extension and mobile to ensure builds are okay.